src/volume.ts

```diff
@@ -98,6 +99,30 @@ export class Volume {
     });
   }
 
+  public static fromSecret(secret: ISecret, options: SecretVolumeOptions = { }): Volume {
+    return new Volume(options.name ?? `secret-${secret.name}`, {
+      secret: {
+        secretName: secret.name,
+        defaultMode: options.defaultMode,
+        optional: options.optional,
+        items: renderItems(options.items),
+      },
+    });
+
+    function renderItems(items?: { [key: string]: PathMapping }) {
+      if (!items) { return undefined; }
+      const result = new Array<k8s.KeyToPath>();
+      for (const key of Object.keys(items).sort()) {
+        result.push({
+          key,
+          path: items[key].path,
+          mode: items[key].mode,
+        });
+      }
+      return result;
+    }
```

Comment:
It looks like this function is the same as the one in `fromConfigMap` - can we move this so there's just one copy that's reused in both places?

Reply:
Created an internal private static function which both Volume.fromSecret and Volume.fromConfigMap. Let me know if I should move this to utils or a standalone function instead. Not sure if the function should be documented since it is not meant for public usage.